dot-license-contracts/contracts/LicenseSale.sol

```diff
@@ -59,63 +85,121 @@ contract LicenseSale is LicenseOwnership {
     }
   }
 
+  function _performRenewal(uint256 _tokenId, uint256 _numCycles) internal {
+    // You cannot renew a non-expiring license
+    // ... but in what scenario can this happen?
+    // require(licenses[_tokenId].expirationTime != 0);
```

Comment:
Perhaps this is better as a note that it's checked in the public function that calls this internal function rather than just "how can this happen?"

Reply:
Agreed -- this is a TODO. I need to decide if this check is worth including.